upup/pkg/fi/cloudup/openstack/floatingip.go

```diff
@@ -137,3 +137,39 @@ func (c *openstackCloud) ListL3FloatingIPs(opts l3floatingip.ListOpts) (fips []l
 	}
 	return fips, nil
 }
+
+func (c *openstackCloud) DeleteFloatingIP(id string) (err error) {
+
+	done, err := vfs.RetryWithBackoff(writeBackoff, func() (bool, error) {
+		err = l3floatingip.Delete(c.ComputeClient(), id).ExtractErr()
+		if err != nil {
+			return false, fmt.Errorf("Failed to delete floating ip %s: %v", id, err)
+		}
+		return true, nil
+	})
+	if !done {
+		if err == nil {
+			err = wait.ErrWaitTimeout
+		}
+		return err
+	}
+	return err
+}
+
+func (c *openstackCloud) DeleteL3FloatingIP(id string) (err error) {
+
+	done, err := vfs.RetryWithBackoff(writeBackoff, func() (bool, error) {
+		err = l3floatingip.Delete(c.NetworkingClient(), id).ExtractErr()
+		if err != nil {
+			return false, fmt.Errorf("Failed to delete L3 floating ip %s: %v", id, err)
+		}
+		return true, nil
+	})
+	if !done {
+		if err == nil {
```

Comment:
I’d join these if clauses and return the err generally. Doesn’t change functionality. 

if !done && err == nil { ... } 
return err

Reply:
If I were to join the clauses my code would look like this
```
if !done && err == nil {
  // This is a regular error
} else if !done {
 // This is a timeout
}
// return success
```
This is actually how its done through-out kops.

What I am suggesting here is that we care more about a `err == nil` condition and check 1 condition instead of two each time we process this section, as each time `err !=nil` is true `!done` is also true (however not vice versa).

Still if you think doing this for consistencies sake of if it looks cleaner I'd be happy to.

Reply:
I mean that’d make sense if you act out something when it’s a timeout error. But no matter if it’s a timeout or a regular error, it’s an error stored in err which you return. So for go this doesn’t matter to be separated, the only special case to be handled is to set err to a timeout error if that’s what happened. Reason enough to compress this for me.

Reply:
you're right.  I was confusing this with the methods that return a pointer and error.